src/Binary/binarySerializer.h

```diff
@@ -15,6 +41,10 @@ class BinarySerializer : public ::Meta::MetaVisitor {
     MetaFile* file;
     BinaryWriter heapWriter;
     BinaryTypeEncodingSerializer typeEncodingSerializer;
+    
+    std::vector<CachedSignature> cachedSignatures;
```

Comment:
Wouldn't it be more efficient to use `std::unordered_map<std::vector<FFIType>, MetaFileOffset>` instead of vector? This way the lookup of a signature will be much faster, avoiding the need for iterating over the vector. However, this requires introducing `hash` and `isEqualTo` operations for `std::vector<FFIType>` which IMO should not be so difficult.